### ISSUE TITLE
new Function(functionBody) gets a function while expection a string.

### ISSUE THREAD

**Issue by simoami** [vvo/chainit#10]
index.js: Line 82

```
  Object.keys(Constructor)
    .forEach(function(name) {
      Chain[name] = new Function(Constructor[name]); // <- this throws an error 
    });
```

The error is:

```
SyntaxError: Unexpected token (
    at Object.Function (<anonymous>)
    at /Users/username/examples/node_modules/chainit/index.js:82
```

To reproduce:

```
var Class = function() {};
Class.prototype.test = function() {};
Class.extend = function() {};

var NewClass = chainit(Class);
```

**Comment by christian-bromann** [comment]
This function assignment:

``` js
Class.extend = function() {};
```

gets resolved into:

``` js
new Function(function() {
})
```

which doesn't work since you cannot pass anonymous function expressions as argument into `new Function`. To fix it just replace

```
Class.extend = function() {};
```

with

```
Class.extend = function extend() {};
```

I will implement a fix later that will handles such situations and will explain it in more detail.
Thank you!

**Comment by simoami** [comment]
Thanks for the quick feedback!

**Comment by christian-bromann** [comment]
As you may know, there are two types of function expressions:

**named**

``` js
var func = function myFunc() {}
```

**anonymous**

``` js
var func = function() {}
```

But there is only one type of function declaration:

``` js
function func() {}
```

By passing an anonymous function expression into the function constructor the actual value of the parameter within the constructor would be

``` js
function() {}
```

which is neither a function expression nor a function declaration. It's just a syntax error. I don't know why `chainit` has assigned static functions to the chain via `new Function` before, but I removed it and just assigning the actual value now. I added [some tests](https://github.com/vvo/chainit/blob/master/test/chain.js#L395) to make sure that static and prototype function get chained properly.

fixed and published (`v2.1.0`)
